engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -372,6 +413,33 @@ public String toString() {
 		return sb.toString();
 	}
 
+	private static boolean buildDisableAlreadyValidatedBeanTracking(ValidationOperation validationOperation, BeanMetaData<?> rootBeanMetaData,
+			Optional<ExecutableMetaData> executableMetaData) {
+		switch ( validationOperation ) {
+			case BEAN_VALIDATION:
+			case PROPERTY_VALIDATION:
+			case VALUE_VALIDATION:
+				// note that in the case of property and value validation, we are considering the root bean, whereas we
+				// could consider the bean of the property or the value. We don't really have the info here though so it
+				// will do for now.
+				return !rootBeanMetaData.hasCascadingProperties();
```

Comment:
Ok, this should work. I still believe it'd have been simpler if `hasCascadables()` would be exposed on `Validatable` (all three objects you check here implement that interface, so there could be a single method for it). But it's fine this way, too.